dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java

```diff
@@ -1,85 +1,318 @@
 package com.dotcms.cluster.bean;
 
+import java.io.Serializable;
 import java.util.Date;
+import javax.annotation.Generated;
 
 
-public class Server {
-	private String serverId;
-	private String clusterId;
-	private String ipAddress;
-	private String host;
-	private String name;
-	private Integer esTransportTcpPort;
-	private Integer esNetworkPort;
-	private Integer esHttpPort;
-	private Integer cachePort;
-	private Date lastHeartBeat;
-	private String key;
+public class Server implements Serializable {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+	public final String serverId;
+	public final  String clusterId;
+	public final  String ipAddress;
+	public final  String host;
+	public final  String name;
+	public final  Integer esTransportTcpPort;
+	public final  Integer esNetworkPort;
+	public final  Integer esHttpPort;
+	public final  Integer cachePort;
+	public final  Date lastHeartBeat;
```

Comment:
needs to be private to insure immutability

Reply:
but they are final once the class is built

Reply:
applies for the Date, which is mutable. Nothing prevents doing this:
Date myDate = server.lastHeartBeat;
myDate.setTime(millis); 
which effectively modifies the state of "server"